tests/models/test_default_evaluator.py

```diff
@@ -1173,3 +1173,41 @@ def add_feature(df):
             "shap_feature_importance_plot_on_data_iris.png",
             "shap_summary_plot_on_data_iris.png",
         }
+
+
+@pytest.mark.parametrize("prefix", ["train_", None])
+@pytest.mark.parametrize("log_metrics_with_dataset_info", [True, False])
+def test_evaluation_metric_name_configs(prefix, log_metrics_with_dataset_info):
+    X, y = load_iris(as_frame=True, return_X_y=True)
+    with mlflow.start_run() as run:
+        model = LogisticRegression()
+        model.fit(X, y)
+        model_info = mlflow.sklearn.log_model(model, "model")
+        result = evaluate(
+            model_info.model_uri,
+            X.assign(target=y),
+            model_type="classifier" if isinstance(model, LogisticRegression) else "regressor",
+            targets="target",
+            dataset_name="iris",
+            evaluators="default",
+            evaluator_config={
+                "metric_prefix": prefix,
+                "log_metrics_with_dataset_info": log_metrics_with_dataset_info,
+            },
```

Comment:
@yxiong @jimmyxu-db Here's an example of how the new config arguments can be used to set a prefix and disable metric logging with dataset info